app/src/main/java/rak/pixellwp/cycling/models/ColorCyclingImage.kt

```diff
@@ -93,23 +76,14 @@ class ColorCyclingImage(img: ImageJson) : PaletteImage {
     }
 
     private fun drawImage() {
-        val p = Paint()
-        var j = 0
-        for (y in 0 until height){
-            for (x in 0 until width) {
-                p.color = palette.colors[pixels[j]]
-                canvas.drawPoint(x.toFloat(), y.toFloat(), p)
-                j++
-            }
+        for (j in 0 until height*width){
+            rawPixels[j] = palette.colors[pixels[j]]
         }
     }
 
     private fun drawOptimizedImage(){
-        val p = Paint()
         for (pixel in optimizedPixels){
-            p.color = palette.colors[pixels[pixel.index]]
-            canvas.drawPoint(pixel.x, pixel.y, p)
+            rawPixels[(pixel.second*width)+pixel.first]=palette.colors[pixels[pixel.third]]
```

Comment:
If you destructure the pixel in the loop you can do a slightly nicer access method here: `for ((x, y, j) in optimizedPixels){`
            
```suggestion
            rawPixels[(y*width)+x]=palette.colors[pixels[j]]
```